addons/account/models/account_move.py

```diff
@@ -486,12 +486,13 @@ def default_get(self, fields):
 
         #compute the default credit/debit of the next line in case of a manual entry
         balance = 0
-        for line in self._context['line_ids']:
-            if line[2]:  # in case of command 0: add a record with values
-                balance += line[2].get('debit', 0) - line[2].get('credit', 0)
-            elif line[0] == 2:  # line has been deleted
-                line_obj = self.browse(line[1])
-                balance -= line_obj.debit - line_obj.credit
+        for line in self.move_id.resolve_2many_commands(
+                'line_ids', self._context['line_ids'], fields=['credit', 'debit']):
+            balance += line.get('debit', 0) - line.get('credit', 0)
+        # if we are here, line_ids is in context, so journal_id should also be.
+        currency = self._context.get('journal_id') and self.browse(self._context['journal_id']).company_id.currency_id
```

Comment:
Looks wrong. `self` is "account.move.line", not "account.journal"
```suggestion
        currency = self._context.get('journal_id') and self.env["account.journal"].browse(self._context['journal_id']).company_id.currency_id
```

This actually break the forward-port to 12.0: http://runbot.odoo.com/runbot/build/596787

Reply:
It is wrong. https://github.com/odoo/odoo/pull/35939